# Anti-AH in Aquaman 2 of Change.org

In [7]:
cd "../../Change.org - Data"

D:\test\mapbox-android-sample-master\src\New folder\Change.org - Data


In [10]:
import os
import pandas as pd

# all petitions comments

In [17]:
data = [i for i in os.listdir() if i.split('.')[-1] == "csv"]

In [18]:
data

['all_petitions_comments_20211026-132241.csv',
 'all_petitions_comments_20211112-095033.csv',
 'all_petitions_comments_4_months.csv',
 'all_petitions_milestones_20211026-130816.csv',
 'all_petitions_milestones_20211112-093758.csv',
 'change_org_comments_petition_slug_hollywood-boycott-aquaman-2-amber-heard_at_20211005-210520_limit_1000_offset_0_delay_ms_100.csv',
 'change_org_petitions_Amber_Heard_at_20211005-222550_limit_1000_offset_0_delay_ms_500 (1).csv',
 'change_org_petitions_Amber_Heard_at_20211005-222550_limit_1000_offset_0_delay_ms_500 (2).csv',
 'change_org_petitions_Amber_Heard_at_20211005-222550_limit_1000_offset_0_delay_ms_500.csv',
 'change_org_petition_slug_united-states-supreme-court-put-amber-heard-behind-bars_at_20210821-084803_max_number_signatures_30 (1).csv']

## petitions comments

In [24]:
df_petitions_comments_1 = pd.read_csv("all_petitions_comments_20211026-132241.csv")
df_petitions_comments_2 = pd.read_csv("all_petitions_comments_20211112-095033.csv")

In [25]:
df_petitions_comments_1.head(1)

index         id                                            comment  \
0      0  810211260  Johnny Depp is Innocent, Amber Heard is a lier...   

   parent_id  children_count commentable_type  commentable_id  likes     role  \
0          0               0            Event        14986339      1  comment   

             created_at  ...  user.photo.sizes.large.size.width  \
0  2020-12-07T02:33:21Z  ...                              400.0   

   user.photo.sizes.large.size.height  \
0                               400.0   

                         user.photo.sizes.xlarge.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   

  user.photo.sizes.xlarge.processing user.photo.sizes.xlarge.size.width  \
0                              False                              800.0   

  user.photo.sizes.xlarge.size.height user.roles  \
0                               800.0         []   

                             commentable_entity.slug  \
0  warner-brothers-save-johnny-depp-as-gellert-gr...   

                                      petition__slug status_manual  
0  warner-brothers-save-johnny-depp-as-gellert-gr...       against  

[1 rows x 45 columns]

In [22]:
df_petitions_comments.shape

(17245, 45)

In [21]:
df_petitions_comments_2.head(1)

index         id                                            comment  \
0      0  810211260  Johnny Depp is Innocent, Amber Heard is a lier...   

   parent_id  children_count commentable_type  commentable_id  likes     role  \
0          0               0            Event        14986339      1  comment   

             created_at  ...  user.photo.sizes.large.size.width  \
0  2020-12-07T02:33:21Z  ...                              400.0   

   user.photo.sizes.large.size.height  \
0                               400.0   

                         user.photo.sizes.xlarge.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   

  user.photo.sizes.xlarge.processing user.photo.sizes.xlarge.size.width  \
0                              False                              800.0   

  user.photo.sizes.xlarge.size.height user.roles  \
0                               800.0         []   

                             commentable_entity.slug  \
0  warner-brothers-save-johnny-depp-as-gellert-gr...   

                                      petition__slug status_manual  
0  warner-brothers-save-johnny-depp-as-gellert-gr...       against  

[1 rows x 45 columns]

In [23]:
df_petitions_comments_2.shape

(27238, 45)

if we conactenated two dfs we shold get 27238 as the data we have scrapped twice
- at 2021-10-26
- 2021-11-26

In [34]:
df_petitions_comments = pd.concat([df_petitions_comments_1, df_petitions_comments_2]).drop_duplicates().drop(columns="index").reset_index(drop=True)

In [35]:
df_petitions_comments.shape

(27238, 44)

In [42]:
pd.options.display.max_columns = 50
df_petitions_comments.head(2)

id                                            comment  parent_id  \
0  810211260  Johnny Depp is Innocent, Amber Heard is a lier...          0   
1  809954316  She's an evil abuser. Justice for JD! She shou...          0   

   children_count commentable_type  commentable_id  likes     role  \
0               0            Event        14986339      1  comment   
1               0            Event        14986339      0  comment   

             created_at  deleted_at     user.id   user.slug user.first_name  \
0  2020-12-07T02:33:21Z         NaN   577958717   577958717           Cesar   
1  2020-12-01T19:45:00Z         NaN  1166768300  1166768300          Brandi   

  user.last_name user.short_display_name user.display_name      user.city  \
0        Ramirez                   Cesar     Cesar Ramirez      Santa Ana   
1        Granger                  Brandi    Brandi Granger  Missouri City   

  user.state_code user.country_code user.website user.locale user.description  \
0              CA                US          NaN       en-US              NaN   
1              TX                US          NaN       en-US              NaN   

   user.photo.id                               user.photo.url  \
0    145246649.0  photos/7/uq/dv/QVuqDvrhjPwmlbz-fullsize.jpg   
1            NaN                                          NaN   

                          user.photo.sizes.small.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   
1                                                NaN   

  user.photo.sizes.small.processing  user.photo.sizes.small.size.width  \
0                             False                               48.0   
1                               NaN                                NaN   

   user.photo.sizes.small.size.height  \
0                                48.0   
1                                 NaN   

                         user.photo.sizes.medium.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   
1                                                NaN   

  user.photo.sizes.medium.processing  user.photo.sizes.medium.size.width  \
0                              False                               128.0   
1                                NaN                                 NaN   

   user.photo.sizes.medium.size.height  \
0                                128.0   
1                                  NaN   

                          user.photo.sizes.large.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   
1                                                NaN   

  user.photo.sizes.large.processing  user.photo.sizes.large.size.width  \
0                             False                              400.0   
1                               NaN                                NaN   

   user.photo.sizes.large.size.height  \
0                               400.0   
1                                 NaN   

                         user.photo.sizes.xlarge.url  \
0  //assets.change.org/photos/7/uq/dv/QVuqDvrhjPw...   
1                                                NaN   

  user.photo.sizes.xlarge.processing  user.photo.sizes.xlarge.size.width  \
0                              False                               800.0   
1                                NaN                                 NaN   

   user.photo.sizes.xlarge.size.height user.roles  \
0                                800.0         []   
1                                  NaN         []   

                             commentable_entity.slug  \
0  warner-brothers-save-johnny-depp-as-gellert-gr...   
1  warner-brothers-save-johnny-depp-as-gellert-gr...   

                                      petition__slug status_manual  
0  warner-brothers-save-johnny-depp-as-gellert-gr...       against  
1  warner-brothers-save-johnny-depp-as-gellert-gr...       against

In [58]:
df_petitions_comments[df_petitions_comments["commentable_entity.slug"].str.contains('aquaman')]["commentable_entity.slug"].unique()

array(['hollywood-boycott-aquaman-2-amber-heard',
       'queremos-que-deborah-ann-woll-reemplace-a-amber-heard-como-mera-en-aquaman-2',
       'warner-brothers-replace-actress-amber-heard-in-aquaman-2',
       'us-remove-amber-heard-from-aquaman',
       'warner-brothers-replace-mera-with-nanaue-in-aquaman',
       'warner-brothers-to-get-amber-heard-fired-in-aquaman',
       'warner-brothers-drop-amber-heard-from-aquaman-2',
       'warner-bros-fire-aquaman-2-producer-peter-safran',
       'warner-brothers-emilia-clarke-to-replace-amber-heard-on-aquaman',
       'dc-comics-replace-amber-heard-with-emilia-clarke-as-mera-in-aquaman-2',
       'warner-brothers-support-amber-heard-in-aquaman-2',
       'support-amber-heard-in-aquaman-2-her-life-and-public-safety-to-save-lives-warner-bros-mera-loreal-united-nations-aclu',
       'warner-brothers-johnny-depp-to-replace-amber-heard-as-mera-in-aquaman-2',
       'warner-brothers-remove-amber-heard-as-mera-for-aquaman-2',
       'warner-broth

In [60]:
pd.read_csv(
    "change_org_petitions_Amber_Heard_at_20211005-222550_limit_1000_offset_0_delay_ms_500.csv"
)

_index      _type                    _id    _score  \
0    petitions-en  petitions  20569720::Amber Heard  7.115500   
1    petitions-en  petitions  26062476::Amber Heard  6.871536   
2    petitions-en  petitions  25704328::Amber Heard  6.744285   
3    petitions-en  petitions  25752458::Amber Heard  6.438685   
4    petitions-en  petitions  13641580::Amber Heard  6.325689   
..            ...        ...                    ...       ...   
288  petitions-en  petitions   4373742::Amber Heard  0.669221   
289  petitions-en  petitions  23141324::Amber Heard  0.633609   
290  petitions-en  petitions  13269107::Amber Heard  0.567439   
291  petitions-en  petitions               13392951  0.507148   
292  petitions-en  petitions               14168553  0.453598   

     _source__total_signature_count _source__created_locale  \
0                             23844                   en-US   
1                             13594                   en-US   
2                             34486                  es-419   
3                             38987                   en-US   
4                             11460                   ru-RU   
..                              ...                     ...   
288                               5                   en-US   
289                              12                   en-US   
290                               6                   en-GB   
291                               8                   en-US   
292                               7                  es-419   

    _source__first_name  _source__weekly_signature_count  \
0                 Linda                              324   
1              Kasandra                               77   
2                Alexis                              116   
3               Dominic                               31   
4                   Ksu                                5   
..                  ...                              ...   
288           Michael S                                0   
289              Hunter                                0   
290                 Tom                                0   
291             Crystal                                0   
292               Bruce                                1   

    _source__recent_activity_date  _source__is_pledge  \
0            2020-03-01T07:41:24Z               False   
1            2021-01-14T22:55:30Z               False   
2            2020-11-07T02:37:54Z               False   
3            2021-05-15T19:11:39Z               False   
4            2019-11-16T18:31:44Z               False   
..                            ...                 ...   
288          2015-09-28T21:04:19Z               False   
289          2020-06-28T21:44:07Z               False   
290          2018-06-09T01:51:59Z               False   
291          2018-07-19T22:06:26Z               False   
292          2019-01-12T14:41:48Z               False   

                        _source__targeting_description   _source__created_at  \
0                                            Hollywood  2020-03-01T07:41:24Z   
1           United States Supreme Court, UK Parliament  2020-11-28T03:46:49Z   
2    PEople with common sense, Johnny Depp, Warner ...  2020-11-07T01:14:06Z   
3           dc, Warner Bros, Amber Heard, danny devito  2020-11-09T15:57:43Z   
4                                          Johnny Depp  2018-10-05T12:57:38Z   
..                                                 ...                   ...   
288                                         Tim Butler  2015-09-28T21:04:19Z   
289  U​.​S. House of Representatives, U​.​S. Senate...  2020-06-28T21:44:07Z   
290                                     Boag's Brewery  2018-06-09T01:51:59Z   
291                                  darnajacpulch1986  2018-07-19T22:06:26Z   
292                                             Rainey  2019-01-12T14:41:48Z   

     _source__user_id  _source__victory _source__victory_date  \
0          1046163930             False  

In [64]:
df_all_petitions_milestones = pd.read_csv(
    "all_petitions_milestones_20211112-093758.csv"
)

In [69]:
df_all_petitions_milestones[df_all_petitions_milestones.petition__slug.str.contains("aquaman")]

index               __typename        id                 createdAt  \
55        0  MilestonePetitionUpdate  28302775  2021-01-04T15:37:35.000Z   
56        1  MilestonePetitionUpdate  28134625  2020-11-28T04:07:26.000Z   
57        2  MilestonePetitionUpdate  28098798  2020-11-20T20:39:26.000Z   
58        3  MilestonePetitionUpdate  27516465  2020-08-14T00:09:01.000Z   
59        4  MilestonePetitionUpdate  25840587  2020-03-04T08:19:51.000Z   
...     ...                      ...       ...                       ...   
1946      2  MilestonePetitionUpdate  28168398  2020-12-04T17:08:52.000Z   
1947      3  MilestonePetitionUpdate  28132791  2020-11-27T18:35:05.000Z   
1948      4  MilestonePetitionUpdate  28102312  2020-11-21T14:56:38.000Z   
1949      5  MilestonePetitionUpdate  28050180  2020-11-11T13:53:39.000Z   
1950      6  MilestonePetitionUpdate  28049586  2020-11-11T11:20:45.000Z   

      signatureCount                                     petition__slug title  \
55             100.0                 us-remove-amber-heard-from-aquaman   NaN   
56              50.0                 us-remove-amber-heard-from-aquaman   NaN   
57              25.0                 us-remove-amber-heard-from-aquaman   NaN   
58              10.0                 us-remove-amber-heard-from-aquaman   NaN   
59               5.0                 us-remove-amber-heard-from-aquaman   NaN   
...              ...                                                ...   ...   
1946           100.0  warner-brothers-remove-amber-heard-from-aquama...   NaN   
1947            50.0  warner-brothers-remove-amber-heard-from-aquama...   NaN   
1948            25.0  warner-brothers-remove-amber-heard-from-aquama...   NaN   
1949            10.0  warner-brothers-remove-amber-heard-from-aquama...   NaN   
1950             5.0  warner-brothers-remove-amber-heard-from-aquama...   NaN   

     description photo video embeddedMedia  dmResponse status_manual  
55           NaN   NaN   NaN           NaN         NaN       against  
56           NaN   NaN   NaN           NaN         NaN       against  
57           NaN   NaN   NaN           NaN         NaN       against  
58           NaN   NaN   NaN           NaN         NaN       against  
59           NaN   NaN   NaN           NaN         NaN       against  
...          ...   ...   ...           ...         ...           ...  
1946         NaN   NaN   NaN           NaN         NaN       against  
1947         NaN   NaN   NaN           NaN         NaN       against  
1948         NaN   NaN   NaN           NaN         NaN       against  
1949         NaN   NaN   NaN           NaN         NaN       against  
1950         NaN   NaN   NaN           NaN         NaN       against  

[762 rows x 13 columns]